src/lesson2-homework.ts

```diff
@@ -1,51 +1,63 @@
-// Create a function which uses tuple type to calculate the distance between two points in 2D space
-function excercise4() {
-  // TODO: declare two variables of type tuple, each with two numbers
-  // TODO: assign values to the variables (1,1) and (4,5)
-  // TODO: create a function which calculates the distance between two points in 2D space
-  function distance(p1: [number, number], p2: [number, number]): number {
-    const x1 = 0; // TODO: replace with the first element of p1
-    const y1 = 0; // TODO: replace with the second element of p1
-    const x2 = 0; // TODO: replace with the first element of p2
-    const y2 = 0; // TODO: replace with the second element of p2
-    // TODO: calculate the distance
-    return 0;
+function excercise4(): void {
+  let x: [number, number] = [1, 1];
+  let y: [number, number] = [4, 5];
+  function calculateDistance(
+    p1: [number, number],
+    p2: [number, number]
+  ): number {
+    const x1 = p1[0];
+    const y1 = p1[1];
+    const x2 = p2[0];
+    const y2 = p2[1];
+    const distance = Math.sqrt(Math.pow(x2 - x1, 2) + Math.pow(y2 - y1, 2));
+    return distance;
   }
-  // TODO: call the function and print the result to console
+  console.log(calculateDistance(x, y));
 }
-// TODO: compile and run the code
+
 excercise4();
 
-// Create a function which uses type alias to calculate the distance between two points in 2D space - points are objects with x and y properties
-function excercise5() {
-  // TODO: declare a type alias for a point in 2D space (TPoint) - object with x and y properties
-  // TODO: declare two variables of type TPoint
-  // TODO: assign values to the variables (1,1) and (4,5)
-  // TODO: create a function which calculates the distance between two points in 2D space
-  type TPoint = { /* replace  with your code */ x: "" };
-  function distance(p1: TPoint, p2: TPoint): number {
-    const x1 = 0; // TODO: replace with the first element of p1
-    const y1 = 0; // TODO: replace with the second element of p1
-    const x2 = 0; // TODO: replace with the first element of p2
-    const y2 = 0; // TODO: replace with the second element of p2
-    // TODO: use distructuring to get x and y from p1 and p2
-    // TODO: calculate the distance
-    return 0;
+function excercise5(): void {
+  type TPoint = {
+    x: number;
+    y: number;
+  };
+  const coordinate1: TPoint = { x: 1, y: 1 };
+  const coordinate2: TPoint = { x: 4, y: 5 };
+  function calculateDistance(p1: TPoint, p2: TPoint): number {
+    const x1 = p1.x;
+    const y1 = p1.y;
+    const x2 = p2.x;
+    const y2 = p2.y;
+    const distance = Math.sqrt(Math.pow(x2 - x1, 2) + Math.pow(y2 - y1, 2));
```

Comment:
// TODO: use distructuring to get x and y from p1 and p2 
this requirement is not implemented